File: Backtracking/78_subsets/78.cpp
#include <vector>
using namespace std;

class Solution {
public:
    
    vector<int> copy(vector<int> set, int num) {
        vector<int> copy(set.size() + 1, 0);
        for (int i = 0; i < set.size(); i++) {
            copy[i] = set[i];
        }
        copy[set.size()] = num;
        return copy;
    }
    
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = nums.size();
        
        vector<vector<int>> subsets = {{}};
        int size;
        
        for (int i = 0; i < n; i++) {
            size = subsets.size();
            for (int j = 0; j < size; j++) {
                subsets.push_back(copy(subsets[j], nums[i]));
            }
        }
        return subsets;
    }
};

File: Backtracking/78_subsets/78.rs


impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut result: Vec<Vec<i32>> = vec!(vec!());
        let mut size: usize = 1;
        for num in nums.iter() {
            for i in 0..size {
                let mut new_set: Vec<i32> = result[i].clone();
                new_set.push(num.clone());
                result.push(new_set);
            }
            size = result.len();
        }
        return result;
    }
}